railgun/internal/proxy/clientgo_cached_proxy_resolver_test.go

```diff
@@ -149,8 +149,27 @@ func TestCaching(t *testing.T) {
 	t.Log("flushing the cache state to kong admin api")
 	proxy.syncTicker.Reset(time.Millisecond * 50)
 
-	t.Logf("waiting for kong admin api updates to synchronize")
-	assert.Eventually(t, func() bool { return fakeKongAdminUpdateCount() == len(testObjects) }, time.Second*5, time.Millisecond*50)
+	t.Logf("ensuring that only a single update to the backend was performed, but that all cache objects are accounted for")
+	assert.Eventually(t, func() bool {
+		return fakeKongAdminUpdateCount() == 1 && len(proxy.cache.IngressV1.List()) == len(testObjects)
+	}, time.Second*5, time.Millisecond*50)
+
+	t.Log("freezing updates to the cache again")
+	fakeKongAdminUpdateCount(0) // reset the test counter
+	proxy.syncTicker.Reset(time.Minute * 3)
+
+	t.Log("deleting all the objects from the cache")
+	for _, testObj := range testObjects {
+		require.NoError(t, proxy.DeleteObject(testObj))
+	}
+
+	t.Log("flushing the cache state to kong admin api again")
+	proxy.syncTicker.Reset(time.Millisecond * 50)
```

Comment:
question: if our cache is already empty, are you also triggering deleting ingress from kong admin api ?

Reply:
Yes, the cache gets built using `deckgen` into one large configuration, the act of removing an object from the cache effectively removes it from the proxy configuration as well, as `/config` updates are last update wins and authoritative declarations of proxy state (not accumulative).

Reply:
@shaneutt  The real question is : I dont see controller sync deleted ingresses through KongAdmin API. Here is current flow:
Ingresses deleted from cluster
Ingress objects deleted from cache
Cache sync empty data from cache, how does controller notify proxy about deletion ?

Correct me if I am wrong.

Reply:
the controller _doesn't notify the proxy about deletion_, the sendconfig mechanism does a _complete replacement of all proxy configuration on each update_. When components are "deleted" what actually happens is that a new configuration is POST'ed without them present, and that has an equivalent effect to "deleting" them.

To put it in other words: all the services, routes, and other objects that you want to be present in the proxy have to be provided in one large configuration, which is POST'd to `/config` on _every change_. Removing resources from this configuration and re-posting the entire config with them gone is how we "delete" resources.

Reply:
I will create a followup issue about this.

Reply:
created https://github.com/Kong/kubernetes-ingress-controller/issues/1581

Reply:
Ok, we can follow up in there.